pkg/cmd/roachtest/tests/mixed_version_multi_region.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestutil/mixedversion"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/spec"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"golang.org/x/exp/maps"
+)
+
+func registerMultiRegionMixedVersion(r registry.Registry) {
+	regionToZones := map[string][]string{
+		"us-east1":        {"us-east1-b"},
+		"us-west1":        {"us-west1-b"},
+		"europe-west2":    {"europe-west2-b"},
+		"europe-central2": {"europe-central2-b"},
+	}
+
+	regions := maps.Keys(regionToZones)
+	var zones []string
+	for _, zs := range maps.Values(regionToZones) {
+		zones = append(zones, zs...)
+	}
+
+	const (
+		nodesPerRegion                  = 20
+		backgroundWarehousesPerRegion   = 30
+		mixedVersionWarehousesPerRegion = 50
```

Comment:
These seem fairly low for a region that has 20 nodes. I'm curious how these were chosen?

Reply:
They were chosen "arbitrarily" 😄. I didn't want the cluster to be overloaded as we plan to add sqlancer (which will take most of the cluster's capacity, I'm assuming). That said, we can increased these numbers for now and lower them in the future as well. I'll play around with them until I reach a workload that will reach ~50% CPU utilization.

Reply:
Makes sense, thanks. During the cloud report, we seemed to have found a way to _linearly_ scale cpu utilization via `tpcc-nowait` by increasing/decreasing connections and warehouses per vCPUs. Unfortunately, I wasn't able to reproduce it with 24.1.0. In any case, there are other ways to scale cpu utilization, albeit artificial, e.g., `stress-ng`.